src/coreclr/ToolBox/superpmi/superpmi/neardiffer.cpp

```diff
@@ -1103,6 +1104,23 @@ bool NearDiffer::compare(MethodContext* mc, CompileResult* cr1, CompileResult* c
                      &coldCodeBlock_2, &roDataBlock_2, &orig_hotCodeBlock_2, &orig_coldCodeBlock_2,
                      &orig_roDataBlock_2);
 
+    if (GetSpmiTargetArchitecture() == SPMI_TARGET_ARCHITECTURE_ARM64)
```

Comment:
Please add a comment here about why this is required, and only for arm64, and generally describing what it is doing.